### PR TITLE
Improve the e2e test cases run

### PR DESCRIPTION
**Pull request by bells17**
Shortened CI time by reducing unnecessary test runs.

ref:
- https://github.com/topolvm/topolvm/pull/545

### CODE REVIEW

File: e2e/suite_test.go [hunk 1]
```diff
@@ -130,17 +130,19 @@ var _ = BeforeSuite(func() {
...
+		Context("topolvm-node", testNode)
+		Context("scheduler", testScheduler)
+		Context("metrics", testMetrics)
+		Context("publish", testPublishVolume)
+		Context("storage-capacity", testStorageCapacity)
+		Context("ReadWriteOncePod", testReadWriteOncePod)
+		Context("e2e", testE2E)
+		Context("multiple-vg", testMultipleVolumeGroups)
+		Context("lvcreate-options", testLVCreateOptions)
+		Context("thin-provisioning", testThinProvisioning)
```
Comment (nbalacha) [inline e2e/suite_test.go:144]: The CSI sanity tests need to be run for thick deviceclasses as well. 
The thin-provisioning tests on line 144 are run on thin deviceclasses.

Comment (bells17) [reply e2e/suite_test.go:144]: > The CSI sanity tests need to be run for thick deviceclasses as well.

These tests are run on e2e tests other than `thin-snapshot-csi`.

Comment (nbalacha) [reply e2e/suite_test.go:144]: > > The CSI sanity tests need to be run for thick deviceclasses as well.
> 
> These tests are run on e2e tests other than `thin-snapshot-csi`.

I did not understand. Can you please elaborate?

Comment (bells17) [reply e2e/suite_test.go:144]: - The new test patern with `TEST_THIN_DEVICECLASS` parameter was added  by https://github.com/topolvm/topolvm/pull/545
- The difference between whether `TEST_THIN_DEVICECLASS` is true or not is only the parameter change of `GINKGO_SKIP_SNAPSHOT_CSI_TESTS`
- `GINKGO_SKIP_SNAPSHOT_CSI_TESTS` defines tests to skip in CSI Sanity tests
- Other than that, the test is executed in the same way as `e2e-k8s / default`, meaning that the test content other than CSI Sanity is duplicated
- Therefore, tests other than CSI Sanity including `thin-provisioning` can be skipped in `thin-snapshot-csi`

Comment (nbalacha) [reply e2e/suite_test.go:144]: Got it. Thank you.

**Review by nbalacha — APPROVED** [review]

**Review by toshipp — APPROVED** [review]

**Pull request merged**